cmd/aries-agentd/main_test.go

```diff
@@ -2,280 +2,22 @@
-	}{
-		// controller API test
-		{
-			name:               "1: testing get",
-			r:                  newreq("GET", fmt.Sprintf("http://%s/connections/create-invitation", testURL), nil, ""),
-			expectedStatus:     http.StatusOK,
-			expectResponseData: true,
-		},
-
-		// DIDComm inbound API test
-		{
-			name: "200: testing didcomm inbound",
-			r: newreq("POST",
-				fmt.Sprintf("http://%s", testInboundURL),
-				strings.NewReader(`
-							{
-								"@id": "5678876542345",
-								"@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/invitation"
-							}`),
-				"application/didcomm-envelope-enc"),
-			expectedStatus:     http.StatusInternalServerError,
-			expectResponseData: false,
-		},
-	}
-	for _, tt := range tests {
-		resp, err := http.DefaultClient.Do(tt.r)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		defer func() {
-			e := resp.Body.Close()
-			if e != nil {
-				panic(err)
-			}
-		}()
-
-		require.Equal(t, tt.expectedStatus, resp.StatusCode)
-		if tt.expectResponseData {
-			require.NotEmpty(t, resp.Body)
-			response, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				t.Fatal(err)
-			}
-			require.NotEmpty(t, response)
-			require.True(t, isJSON(response))
-		}
-	}
-}
-
-// isJSON checks if response is json
-func isJSON(res []byte) bool {
-	var js map[string]interface{}
-	return json.Unmarshal(res, &js) == nil
-}
-
-func TestStartAriesDWithoutHost(t *testing.T) {
-	prev := os.Getenv(agentHostEnvKey)
-	defer func() {
-		err := os.Setenv(agentHostEnvKey, prev)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}()
-
-	err := os.Setenv(agentHostEnvKey, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	done := make(chan bool)
-
-	go func() {
-		main()
-		done <- true
-	}()
-
-	select {
-	case res := <-done:
-		require.True(t, res)
-	case <-time.After(5 * time.Second):
-		t.Fatal("agent should fail to start when host address not provided")
-	}
-}
-
-func TestStartAriesWithoutInboundHost(t *testing.T) {
-	prev := os.Getenv(agentHostEnvKey)
-	defer func() {
-		err := os.Setenv(agentHostEnvKey, prev)
-		if err != nil {
-			t.Fatal(err)
-		}
-	}()
-
-	err := os.Setenv(agentHostEnvKey, randomURL())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	prevInboundHost := os.Getenv(agentHTTPInboundEnvKey)
-	defer func() {
-		inboundConfigErr := os.Setenv(agentHTTPInboundEnvKey, prevInboundHost)
-		if inboundConfigErr != nil {
-			t.Fatal(inboundConfigErr)
-		}
-	}()
-
-	err = os.Setenv(agentHTTPInboundEnvKey, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	done := make(chan bool)
-
-	go func() {
-		main()
-		done <- true
-	}()
-
-	select {
-	case res := <-done:
-		require.True(t, res)
-	case <-time.After(5 * time.Second):
-		t.Fatal("agent should fail to start when inbound host address not provided")
-	}
+// Correct behaviour is for main to finish with exit code 0.
+// This test fails otherwise. However, this can't be checked by the unit test framework, so t goes unused.
+func TestCallingWithNoUserArgs(t *testing.T) { //nolint - see above
```

Comment:
typo in comment: so t goes unused.
fn name: TestWithOutUserArgs??

Reply:
It actually wasn't a typo - the "t" is referring to t *testing.T. But it does look confusing. I reworded it to: // This test fails otherwise. However, this can't be checked by the unit test framework. The *testing.T argument is
// only there so that this test gets picked up by the framework but otherwise we don't need it.

Reply:
Done - renamed the test.